CMakeLists.txt

```diff
@@ -326,6 +326,10 @@ include_directories(${STB_DIR})
 set(EXCLUDE_MPQ_SUPPORT TRUE CACHE BOOL "")
 add_compile_definitions(EXCLUDE_MPQ_SUPPORT)
 
+if(CMAKE_SYSTEM_NAME STREQUAL "NintendoSwitch")
+    find_package(SDL2)
+endif()
```

Comment:
running `find_package` ensures `SDL2_INCLUDE_DIRS` is set, and needs to happen before we try to use it. i'm honestly not sure why this isn't an issue for other platforms.